### PR TITLE
Don't pass specific options to the parent constructor

### PR DESCRIPTION
**Pull request by fredj**
Same as #3104

### CODE REVIEW

File: src/ol/layer/tilelayer.js [hunk 1]
```diff
@@ -29,14 +30,16 @@ ol.layer.TileProperty = {
...
+
+  var baseOptions = goog.object.clone(options);
+
+  delete baseOptions.preload;
+  delete baseOptions.useInterimTilesOnError;
+  goog.base(this,  /** @type {olx.layer.LayerOptions} */ (baseOptions));
+
+  this.setPreload(goog.isDef(options.preload) ? options.preload : 0);
+  this.setUseInterimTilesOnError(goog.isDef(options.useInterimTilesOnError) ?
+      options.useInterimTilesOnError : true);
```
Comment (elemoine) [inline src/ol/layer/tilelayer.js]: You could probably do it w/o cloning the options object:

```js
ol.layer.Tile = function(opt_options) {
  var options = goog.isDef(opt_options) ? opt_options : {};
  
  var preload = goog.isDef(options.preload) ? options.preload : 0;
  var useInterimTilesOnError = goog.isDef(options.useInterimTilesOnError) ?
      options.useInterimTilesOnError : true;

  delete options.preload;
  delete options.useInterimTilesOnError;

  goog.base(this, /** @type {olx.layer.LayerOptions} */ (options));

  this.setPreload(preload);
  this.setUseInterimTilesOnError(useInterimTilesOnError);
};
```
(untested)

Comment (fredj) [inline src/ol/layer/tilelayer.js]: +1

Comment (fredj) [inline src/ol/layer/tilelayer.js]: You code works but I suggest to keep the `goog.object.clone`: it is how it's done in the other classes (`ol.layer.Group`, `ol.layer.Vector`, ...) and we will had to change if if/when we have a new non primitive type to handle.

Comment (elemoine) [inline src/ol/layer/tilelayer.js]: > You code works but I suggest to keep the goog.object.clone: it is how it's done in the other classes (ol.layer.Group, ol.layer.Vector, ...)

Ok.



> and we will had to change if if/when we have a new non primitive type to handle.

We'd need to add `delete` statements anyway, even if `clone` is used. No?

Comment (fredj) [inline src/ol/layer/tilelayer.js]: > We'd need to add delete statements anyway, even if clone is used. No?

Correct.

I thought that deleting a non primitive type from an object will delete the copy too:
```javascript
var obj = {
  arr: [1,2,3]
};
var c = obj.arr;
delete obj.arr;
```
Unlike what I was thinking: `c == [1,2,3]`

Comment (elemoine) [inline src/ol/layer/tilelayer.js]: `delete` just removes the property from an object. It does nothing to the object that property refers to.

**Comment by fredj** [comment]
@elemoine do you agree to merge this PR as it?

**Comment by elemoine** [comment]
Yep. Please merge.

**Pull request merged**